micro_sam/sam_annotator/_annotator.py

```diff
@@ -0,0 +1,165 @@
+    """
+
+    def _create_layers(self, segmentation_result):
+        # Add the point layer for point prompts.
+        self._point_labels = ["positive", "negative"]
+        # We need to add dummy data to initialize the point properties correctly.
+        # The two points will be cleared at the end.
+        dummy_data = 2 * [[0.0] * self._ndim]
+        self._point_prompt_layer = self._viewer.add_points(
+            data=dummy_data,
+            name="point_prompts",
+            properties={"label": self._point_labels},
+            edge_color="label",
+            edge_color_cycle=vutil.LABEL_COLOR_CYCLE,
+            symbol="o",
+            face_color="transparent",
+            edge_width=0.5,
+            size=12,
+            ndim=self._ndim,
+        )
+        self._point_prompt_layer.edge_color_mode = "cycle"
+
+        # Add the shape layer for box and other shape prompts.
+        self._viewer.add_shapes(
+            face_color="transparent", edge_color="green", edge_width=4, name="prompts", ndim=self._ndim,
+        )
+
+        # Add the label layers for the current object, the automatic segmentation and the committed segmentation.
+        dummy_data = np.zeros(self._shape, dtype="uint32")
+        self._viewer.add_labels(data=dummy_data, name="current_object")
+        self._viewer.add_labels(data=dummy_data, name="auto_segmentation")
+        self._viewer.add_labels(
+            data=dummy_data if segmentation_result is None else segmentation_result, name="committed_objects"
+        )
+        # Randomize colors so it is easy to see when object committed.
+        self._viewer.layers["committed_objects"].new_colormap()
+
+    def _create_widgets(self, segment_widget, segment_nd_widget, autosegment_widget, commit_widget, clear_widget):
+        self._embedding_widget = widgets.embedding_widget()
+        # Connect the call button of the embedding widget with a function
+        # that updates all relevant layers when the image changes.
+        self._embedding_widget.call_button.changed.connect(self._update_image)
+
+        self._prompt_widget = widgets.create_prompt_menu(self._point_prompt_layer, self._point_labels)
+        self._segment_widget = segment_widget()
+        widget_list = [self._embedding_widget, self._prompt_widget, self._segment_widget]
+
+        if segment_nd_widget is not None:
+            self._segment_nd_widget = segment_nd_widget()
+            widget_list.append(self._segment_nd_widget)
+
+        if autosegment_widget is not None:
+            self._autosegment_widget = autosegment_widget()
+            widget_list.append(self._autosegment_widget)
+
+        self._commit_widget = commit_widget()
+        self._clear_widget = clear_widget()
+        widget_list.extend([self._commit_widget, self._clear_widget])
+
+        # Add the widgets to the container.
+        self.extend(widget_list)
+
+    def _create_keybindings(self):
+        @self._viewer.bind_key("s")
+        def _segmet(viewer):
```

Comment:
```suggestion
        def _segment(viewer):
```

Small typo here - will also need a quick find/replace on any other locations in the code that use this function

Reply:
Nice catch, thanks! And this is just a local function for the key bindings so we don't need to fix it anywhere else. (I also did a search for `segmet` and this is the only place where the typo occurs.)